openssl/src/x509/mod.rs

```diff
@@ -921,6 +929,40 @@ impl X509Extension {
         }
     }
 
+    /// Constructs a new X509 extension value from its OID, whether it's
+    /// critical, and its DER contents.
+    ///
+    /// The extent structure of the DER value will vary based on the
+    /// extension type, and can generally be found in the RFC defining the
+    /// extension.
+    ///
+    /// For common extension types, there are Rust APIs provided in
+    /// `openssl::x509::extensions` which are more ergonomic.
+    pub fn new_from_der(
+        oid: &Asn1ObjectRef,
+        critical: bool,
+        der_contents: &[u8],
+    ) -> Result<X509Extension, ErrorStack> {
+        unsafe {
+            ffi::init();
+            let s = Asn1String::from_ptr(cvt_p(ffi::ASN1_STRING_type_new(
+                Asn1Type::OCTET_STRING.as_raw(),
+            ))?);
+            cvt(ffi::ASN1_STRING_set(
+                s.as_ptr(),
+                der_contents.as_ptr().cast(),
+                der_contents.len().try_into().unwrap(),
+            ))?;
```

Comment:
Should we have this method take an `Asn1OctetStringRef` instead of doing the construction in here?

Reply:
I don't think it has a public constructor, so we'd have to do that first. This seemed simpler.

Reply:
Should be pretty straightforward to add the constructor.

Reply:
Done